python/cuml/test/test_metrics.py

```diff
@@ -160,40 +163,90 @@ def test_rand_index_score(name, nrows):
     assert array_equal(cu_score, cu_score_using_sk)
 
 
-def test_mean_squared_error():
+def test_regression_metrics():
     y_true = np.arange(50, dtype=np.int)
     y_pred = y_true + 1
     assert_almost_equal(mean_squared_error(y_true, y_pred), 1.)
+    assert_almost_equal(mean_squared_log_error(y_true, y_pred),
+                        mean_squared_error(np.log(1 + y_true),
+                                           np.log(1 + y_pred)))
+    assert_almost_equal(mean_absolute_error(y_true, y_pred), 1.)
 
 
 @pytest.mark.parametrize('n_samples', [50, stress_param(500000)])
 @pytest.mark.parametrize('dtype', [np.int32, np.int64, np.float32, np.float64])
-def test_mean_squared_error_random(n_samples, dtype):
+def test_regression_metrics_random(n_samples, dtype):
     if dtype == np.float32 and n_samples == 500000:
         # stress test for float32 fails because of floating point precision
         pytest.xfail()
 
     y_true, y_pred = generate_random_labels(
         lambda rng: rng.randint(0, 1000, n_samples).astype(dtype))
+
     mse = mean_squared_error(y_true, y_pred, multioutput='raw_values')
     skl_mse = sklearn_mse(y_true, y_pred, multioutput='raw_values')
     cp.testing.assert_array_almost_equal(mse, skl_mse, decimal=2)
 
+    mae = mean_absolute_error(y_true, y_pred, multioutput='raw_values')
+    skl_mae = sklearn_mae(y_true, y_pred, multioutput='raw_values')
+    cp.testing.assert_array_almost_equal(mae, skl_mae, decimal=2)
+
+    msle = mean_squared_log_error(y_true, y_pred, multioutput='raw_values')
+    skl_msle = sklearn_msle(y_true, y_pred, multioutput='raw_values')
+    cp.testing.assert_array_almost_equal(msle, skl_msle, decimal=2)
 
-def test_mean_squared_error_at_limits():
+
+def test_regression_metrics_at_limits():
     y_true = np.array([0.], dtype=np.float)
     y_pred = np.array([0.], dtype=np.float)
     assert_almost_equal(mean_squared_error(y_true, y_pred), 0.00, decimal=2)
     assert_almost_equal(mean_squared_error(y_true, y_pred, squared=False),
                         0.00, decimal=2)
+    assert_almost_equal(mean_squared_log_error(y_true, y_pred), 0.00, 2)
+    assert_almost_equal(mean_absolute_error(y_true, y_pred), 0.00, 2)
+
+    err_msg = ("Mean Squared Logarithmic Error cannot be used when targets "
+               "contain negative values.")
+    with pytest.raises(ValueError, match=err_msg):
```

Comment:
I would typically just assert that we raise the right class of error and not check the message directly... if tests get over-specified (covering non-essential behavior), they can make the code harder to change, since each change will require updating a ton of strings and asserts here.. definitely good to check that the exception itself is raised though!

Reply:
Good point, that should help us maintaining it in the future.